### ISSUE TITLE
Save file not generating

### ISSUE THREAD

**Issue by TheSkulled** [suicvne/DOR_Remake#8]
When starting the game, I press create new save.
Then I can type my name.
I can navigate the decks menu.
However, it does not create a file when I press create file. 
When spacebar is pressed nothing happens.

![image](https://user-images.githubusercontent.com/84255874/118382810-419c7e00-b63c-11eb-8ea9-660c6ce80dae.png)

**Comment by Levitikan** [comment]
The same thing happened to me too, but when i looked at my documents folder there was a folder there called DOR Remake saves with a profile that i named it, it just isn't showing when loading saves, have u checked top see of the folder is there. as this is what happens to me
![DOR bug](https://user-images.githubusercontent.com/37230091/118421855-e01af480-b6b9-11eb-98f7-d72d4b73cd55.png)

**Comment by suicvne** [comment]
I was going to respond to this on YouTube.

@TheSkulled 
I had *thought* that I had fixed the Create Save bug already but clearly not. Please attach log files. They're located in
```
C:\Users\<username>\AppData\LocalLow\IgnoreSolutions\DORNetTest
```
The file is Player.log

I'm just still scratching my head as to how this works PERFECTLY on Mac and Linux, but WINDOWS of all things causes the problems. Ugh. This is especially frustrating for me because my dedicated Windows computer is no longer posting.


@Levitikan something like that will happen when it has an error reading saves, i will need a log file for that too so I can figure out exactly what's tripping it out.

**Comment by suicvne** [comment]
#2 should have fixed @TheSkulled 's problem with the Create Save button just flat out not working/not responding. There must be something else going on. 

Please also make sure you both are on version 0.6.8.2 which can be downloaded from itch.io 

UPDATE: The Windows PC posted after being unplugged for 2+ weeks! Time to get some work done bois!!!!!!

**Comment by suicvne** [comment]
0.6.8.3 released on itch.io!
This may fix a lot of the problems you guys are having.

https://ignoresolutions.itch.io/yu-gi-oh-duelists-of-the-roses-remake-beta

**Comment by Levitikan** [comment]
This is unfortuante  and really odd here is my player.log on 0.6.8.3
[Player.log](https://github.com/suicvne/DOR_Remake/files/6497859/Player.log)

**Comment by suicvne** [comment]
Tricky card number 804!!!
Does anyone know off the top of their head what #804 should be? 

I have an 803 and 805 but no 804 from my generated data. Unsure if missing or misnunbered. 

Anyway, thats the crash. Can’t find card #804. I could temp patch that or add #804 and then patch that NullException.

**Comment by suicvne** [comment]
Would you mind posting your save here so I can try it exactly on my build?

**Comment by rjoken** [comment]
> Tricky card number 804!!!
> Does anyone know off the top of their head what #804 should be?

Should be [Tears of the Mermaid](https://yugipedia.com/wiki/Tears_of_the_Mermaid_(DOR))
However, the Wolf Axwielder deck does not contain that card (in the vanilla game). I tested the Maiden of the Aqua deck which _does_ contain that card, and I had the same problem as OP. Also happened with Injection Fairy Lily and Airknight Parshath which contain tears in vanilla too.

Update: ok, none of the decks are letting me create a save file now apart from Robotic Knight (still testing)
Another edit: sometimes i can pick a deck, sometimes i can't. now robotic knight isn't working. i'm guessing it's not tied to the deck leader selection at all based on the log which i will now attach:
[Player.log](https://github.com/suicvne/DOR_Remake/files/6498452/Player.log)

**Comment by TheSkulled** [comment]
I tried it, the save files are being made but the game doesn't seem to be able to locate them. I tried booting it off both of my hard drives and both times it didn't work.

[Player.log](https://github.com/suicvne/DOR_Remake/files/6500146/Player.log)

**Comment by Levitikan** [comment]
Here is my save files, good luck with fixing the issue

[DOR Remake Saves.zip](https://github.com/suicvne/DOR_Remake/files/6500596/DOR.Remake.Saves.zip)

**Comment by suicvne** [comment]
Tears of the Mermaid is already in the game so I’ve got a numbering problem on my end. I’ve also definitely created decks with those leaders in the past so I’m not sure. 

UPDATE tears of the mermaid isnt in the game. the card list skips from 803 to 805. Oops.

**Comment by suicvne** [comment]
I also recently put some janky code to reset the Deck selection carousel so thats probably causing the inconsistencies with the selection

**Comment by suicvne** [comment]
@TheSkulled  that log still had the card 804 error. the game is discovering the saves and starting to read them but it gets tripped up trying to get the 804 card. 

I think the root of all of this is improper numbering

**Comment by suicvne** [comment]
Fixed the card #804 bug. The card existed on disk and in data, which allowed me to assign to the decks, but it wasn't actually added to the MasterCardList that the game references internally. Adding that card to the MasterCardList allowed all 3 of Levitikan's saves to load and show up on my end. 

![SAVES WORKING](https://user-images.githubusercontent.com/861492/118653881-fa39fd00-b7b5-11eb-8bf1-ecf26fb71f17.png)

(don't worry about the "saved with older version", just load the save and it will convert itself)

I'm uploading version 0.6.8.4a to itch.io before I leave for the beach today. Thanks everyone for helping with getting this issue ironed out!


I also just realized my game data is *technically* missing #671.

"Summoned Lord Exodia"

![SummonedLordExodia-DOR-EN-VG](https://user-images.githubusercontent.com/861492/118654107-353c3080-b7b6-11eb-8eed-68c8b04f2542.png)

I didn't even know this was in the game. I'm guessing you're allowed to view this in the gallery after seeing Exodia once?
